lib/ansible/module_utils/network/netconf/netconf.py

```diff
@@ -18,13 +18,37 @@
 #
 import json
 
+from copy import deepcopy
 from contextlib import contextmanager
 
-from ansible.module_utils._text import to_text
+try:
+    from lxml.etree import fromstring, tostring
+except ImportError:
+    from xml.etree.ElementTree import fromstring, tostring
+
+from ansible.module_utils._text import to_text, to_bytes
+from ansible.module_utils.basic import env_fallback
 from ansible.module_utils.connection import Connection, ConnectionError
 from ansible.module_utils.network.common.netconf import NetconfConnection
 
 
+netconf_argument_spec = {}
+
+netconf_top_spec = {
+    'host': dict(removed_in_version=2.11),
+    'port': dict(removed_in_version=2.11, type='int', default=830),
+    'username': dict(fallback=(env_fallback, ['ANSIBLE_NET_USERNAME']), removed_in_version=2.11, no_log=True),
+    'password': dict(fallback=(env_fallback, ['ANSIBLE_NET_PASSWORD']), removed_in_version=2.11, no_log=True),
+    'ssh_keyfile': dict(fallback=(env_fallback, ['ANSIBLE_NET_SSH_KEYFILE']), removed_in_version=2.11, type='path'),
+    'hostkey_verify': dict(removed_in_version=2.11, type='bool', default=True),
+    'look_for_keys': dict(removed_in_version=2.11, type='bool', default=True),
+    'timeout': dict(removed_in_version=2.11, type='int', default=10),
+}
+netconf_argument_spec.update(netconf_top_spec)
```

Comment:
I'm struggling to remember why this is necessary. `netconf_top_spec` is not to my recollection used anywhere outside of here, and without `provider`, building one dictionary, then updating an empty dictionary with its contents seems redundant. Although, `_top_spec` seems redundant even then, so I'm probably missing something here.

Reply:
I referenced this from other network modules in which credential args are part of `module_utils/<network_os>/<network_os>.py`. That's why thought of keeping the same format (easier to remove this code when deprecation cycle ends). 

Though agree that it can be moved in `netconf_config` module code.

Reply:
Fixed, make sense to move it on the netconf_config module as other netconf modules are not using it. Thanks for pointing it out.